app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt

```diff
@@ -36,6 +48,14 @@ class HomeViewModel
 
     suspend fun navigateToUserProfile() = navigateTo(NavigationItem.UserProfile, MY_USER_PROFILE_SUBROUTE)
 
+    fun loadUserAvatar() {
+        viewModelScope.launch {
+            dataStore.avatarAssetId.first()?.let {
+                userAvatar = (getPublicAssetUseCase(it) as PublicAssetResult.Success).asset
+            }
+        }
+    }
```

Comment:
are we handeling here errors ? you just cast it to  PublicAssetResult.Success, which is abit dangerous I think here ;p

Reply:
Good catch. Adding a dummy try/catch clause for now. In any case this smart casts will be soon refactored once we decide on a better wrapping for these use cases results